src/chunk.c

```diff
@@ -737,6 +788,9 @@ ts_chunk_create_table(Chunk *chunk, Hypertable *ht, const char *tablespacename)
 		 */
 		create_toast_table(&stmt.base, objaddr.objectId);
```

Comment:
does toast table need the right ACLs as well?

Reply:
Good point, will check.

Reply:
Doesn't seem like this is necessary. The ACLs to toast tables are always NULL and any permission checking done is done on the "real" table, not the toast table.

Here is from `create_toast_table`:

```c
	toast_relid = heap_create_with_catalog(toast_relname,
										   namespaceid,
										   rel->rd_rel->reltablespace,
										   toastOid,
										   toast_typid,
										   InvalidOid,
										   rel->rd_rel->relowner,
										   rel->rd_rel->relam,
										   tupdesc,
										   NIL,
										   RELKIND_TOASTVALUE,
										   rel->rd_rel->relpersistence,
										   shared_relation,
										   mapped_relation,
										   ONCOMMIT_NOOP,
										   reloptions,
										   false,      /* NOTE[mats]: use_user_acl */
										   true,
										   true,
										   InvalidOid,
										   NULL);
```

Docs for `use_user_acl`:

```
 *	use_user_acl: true if should look for user-defined default permissions;
 *		if false, relacl is always set NULL
```

The `toastOid` passed in is `InvalidOid` (which means "create a new OID") when used from `NewHeapCreateToastTable`, which is what we are calling here.

In addition, `ExecGrant_Relation` (which is the implementation of the grant statement for relations) does not recurse to TOAST tables for changing grants (even though the relid for the toast table is available), which suggest that they are not used.